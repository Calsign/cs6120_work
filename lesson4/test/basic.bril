@main() {
  x: int = const 3;
  y: int = id x;
  x: int = mul x y;
}
