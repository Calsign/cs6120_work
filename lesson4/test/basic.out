{"op":"const","value":3,"dest":"x","type":"int"}  ->  x: 3
{"op":"id","args":["x"],"dest":"y","type":"int"}  ->  x: 3, y: 3
{"op":"mul","args":["x","y"],"dest":"x","type":"int"}  ->  x: ?, y: 3
